File: 2Star AdvLvl/EXUNB.cpp
#include <bits/stdc++.h>
using namespace std;

void tennisTourny() {
    int n;
    cin >> n;
    if (n % 2 == 0)
        cout << "NO" << endl;
    else {
        vector<vector<char>> ans(n, vector<char>(n, '0'));

        for (int i = 0; i < n; i++) {
            for (int j = 1; j <= n / 2; j++) {
                ans[i][(i + j) % n] = '1';
            }
        }

        cout << "YES" << endl;
        for (const vector<char>& line : ans) {
            for (char ch : line) {
                cout << ch;
            }
            cout << endl;
        }
    }
}

int main() {
	// your code goes here
	int t;
	cin >> t;
	while (t--) {
	    tennisTourny();
	}
	return 0;
}

File: 2Star AdvLvl/EXUNB.py
def tennisTourny():
    n = int(input())
    if n % 2 == 0:
        print("NO")
    else:
        ans = [['0'] * n for _ in range(n)]

        for i in range(n):
            for j in range(1, n // 2 + 1):
                ans[i][(i + j) % n] = '1'

        print("YES")
        for line in ans:
            print("".join(line))

t = int(input())
for _ in range(t):
    tennisTourny()
